spaceship/lib/spaceship/client.rb

```diff
@@ -37,71 +42,16 @@ class Client
 
     attr_accessor :available_providers
 
-    # Base class for errors that want to present their message as
-    # preferred error info for fastlane error handling. See:
-    # fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb
-    class BasicPreferredInfoError < StandardError
-      TITLE = 'The request could not be completed because:'.freeze
-
-      def preferred_error_info
-        message ? [TITLE, message] : nil
-      end
-    end
-
-    # Invalid user credentials were provided
-    class InvalidUserCredentialsError < BasicPreferredInfoError; end
-
-    # Raised when no user credentials were passed at all
-    class NoUserCredentialsError < BasicPreferredInfoError; end
-
-    class ProgramLicenseAgreementUpdated < BasicPreferredInfoError
-      def show_github_issues
-        false
-      end
-    end
-
-    # User doesn't have enough permission for given action
-    class InsufficientPermissions < BasicPreferredInfoError
-      TITLE = 'Insufficient permissions for your Apple ID:'.freeze
-
-      def preferred_error_info
-        message ? [TITLE, message] : nil
-      end
-
-      # We don't want to show similar GitHub issues, as the error message
-      # should be pretty clear
-      def show_github_issues
-        false
-      end
-    end
-
-    class UnexpectedResponse < StandardError
-      attr_reader :error_info
-
-      def initialize(error_info = nil)
-        super(error_info)
-        @error_info = error_info
-      end
-
-      def preferred_error_info
-        return nil unless @error_info.kind_of?(Hash) && @error_info['resultString']
-
-        [
-          "Apple provided the following error info:",
-          @error_info['resultString'],
-          @error_info['userString']
-        ].compact.uniq # sometimes 'resultString' and 'userString' are the same value
-      end
-    end
-
-    # Raised when 302 is received from portal request
-    class AppleTimeoutError < BasicPreferredInfoError; end
-
-    # Raised when 401 is received from portal request
-    class UnauthorizedAccessError < BasicPreferredInfoError; end
-
-    # Raised when 500 is received from iTunes Connect
-    class InternalServerError < BasicPreferredInfoError; end
+    # legacy support
+    BasicPreferredInfoError = Spaceship::BasicPreferredInfoError
+    InvalidUserCredentialsError = Spaceship::InvalidUserCredentialsError
+    NoUserCredentialsError = Spaceship::NoUserCredentialsError
+    ProgramLicenseAgreementUpdated = Spaceship::ProgramLicenseAgreementUpdated
+    InsufficientPermissions = Spaceship::InsufficientPermissions
+    UnexpectedResponse = Spaceship::UnexpectedResponse
+    AppleTimeoutError = Spaceship::AppleTimeoutError
+    UnauthorizedAccessError = Spaceship::UnauthorizedAccessError
+    InternalServerError = Spaceship::InternalServerError
```

Comment:
All the errors have been refactored to their own `errors.rb` file and live in the `Spaceship` module now, with these wrappers provided for legacy support